src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -591,12 +591,38 @@ private bool TakeStartLine(SocketInput input)
 
                 scan.Take();
                 begin = scan;
-                var chFound = scan.Seek(' ', '?');
-                if (chFound == -1)
+
+                int chFound = -1;
+                bool needDecode = false;
+                while (true)
```

Comment:
this while true is wierd. At most you're only ever going to call seek twice, correct?
```
chFound = scan.Seek(' ', '?', '%');
if (chFound == %)
{
  needsDecode = true;
  chFound = scan.Seek(' ', '?');
}
```